java/core/src/main/java/co/worklytics/psoxy/impl/RESTApiSanitizerImpl.java

```diff
@@ -191,6 +192,15 @@ Object applyTransform(Transform transform, Object document ) {
                 .map(JsonPath::compile)
                 .collect(Collectors.toList()));
 
+        if (transform.getApplyOnlyWhen() != null) {
+            Object filterResult = JsonPath.compile(transform.getApplyOnlyWhen()).read(document);
```

Comment:
Support apply this transformation only when the condition is satisfied. The scenario is the following:
- For Users, the id needs to be pseudonymized
- For accounts, id is not pseudonymized.

From the query pagination response, it means that:
- If `attributes.type == User` the user rules needs to be applied for transformation
- If `attributes.type == Account` the account rules needs to be applied for transformation

And this is resolved checking the json path expression set no `applyOnlyWhen`, for example:
```
$..records[?(@.attributes.type == \"User\")]
```

So the transformation will be applied if the json path conditional expression is matched with something (aka: there are NO empty results after applying that rule)

Reply:
OK for now, but you're assuming here that compilation of the JSON path transform is cheap.  I'm ok with it, bc should avoid premature optimization.